pre-commit autoupdate

DimitriPapadopoulos opened the pull request:
Depends on #1096 - ruff needs to be updated first.

DimitriPapadopoulos committed:
pre-commit autoupdate

DimitriPapadopoulos commented on _own_version_helper.py:
```diff
@@ -13,7 +13,6 @@
 
 from setuptools import build_meta as build_meta
-
 from setuptools_scm import Configuration
```
This doesn't look good. We do want to separate `setuptools` and  `setuptools_scm` imports, don't we?

RonnyPfannschmidt replied:
good catch - im not sure why it got merged - is the detection going wronf

DimitriPapadopoulos replied:
I will give [`lint.isort.known-local-folder`](https://docs.astral.sh/ruff/settings/#lint_isort_known-local-folder) a try.

DimitriPapadopoulos replied:
Caused by the obsolete version of ruff prior to #1096. Fixed by moving one of the commits to that PR.

DimitriPapadopoulos committed:
Fix mypy [unused-ignore] error

Unused "type: ignore" comment

DimitriPapadopoulos committed:
Apply sp-repo-review rule RF003

src directory doesn't need to be specified anymore (0.6+)

DimitriPapadopoulos committed:
Python 3.13

DimitriPapadopoulos committed:
Ignore ruff/flake8-pytest-style rule PT030

PT030 `pytest.warns(UserWarning)` is too broad, set the `match` parameter or use a more specific warning

RonnyPfannschmidt commented on .pre-commit-config.yaml:
```diff
@@ -1,26 +1,26 @@
     - id: ruff-format
 
 - repo: https://github.com/tox-dev/pyproject-fmt
-  rev: 1.7.0
```
This one needs to be blocked 

The next gen is unacceptable

DimitriPapadopoulos replied:
Is 1.8.0 OK, tough?

RonnyPfannschmidt replied:
I haven't verified whether it inlines entrypoints 

Any version that doesn't can be used

DimitriPapadopoulos replied:
1.8.0 is OK then.

DimitriPapadopoulos commented on pyproject.toml:
```diff
@@ -31,7 +31,6 @@ classifiers = [
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
-  "Programming Language :: Python :: 3.13",
```
Strange, why does pre-commit remove Python 3.13?

DimitriPapadopoulos replied:
Removing this line for now. Will investigate in a different PR.

RonnyPfannschmidt reviewed (approved):
Lovely, thanks
